go/libraries/doltcore/sqle/index/key_builder.go

```diff
@@ -15,16 +15,23 @@
 package index
 
 import (
-	"github.com/dolthub/go-mysql-server/sql/types"
+	"context"
 
 	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
 	"github.com/dolthub/dolt/go/store/pool"
+	"github.com/dolthub/dolt/go/store/prolly/tree"
 	"github.com/dolthub/dolt/go/store/val"
 )
 
-func NewSecondaryKeyBuilder(sch schema.Schema, def schema.Index, idxDesc val.TupleDesc, p pool.BuffPool) (b SecondaryKeyBuilder) {
+// NewSecondaryKeyBuilder creates a new SecondaryKeyBuilder instance that can build keys for the secondary index |def|.
+// The schema of the source table is defined in |sch|, and |idxDesc| describes the tuple layout for the index's keys
+// (index value tuples are not used).
+func NewSecondaryKeyBuilder(sch schema.Schema, def schema.Index, idxDesc val.TupleDesc, p pool.BuffPool, nodeStore tree.NodeStore) (b SecondaryKeyBuilder) {
 	b.builder = val.NewTupleBuilder(idxDesc)
```

Comment:
Seems like an unnamed result param and a struct literal is more idiomatic for these constructor funcs